### PR TITLE
Added minimal cookbook article about the shared flag

### PR DESCRIPTION
**Pull request by wouterj**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes (https://github.com/symfony/symfony/pull/14984) |
| Applies to | 2.8+ |
| Fixed tickets | https://github.com/symfony/symfony-docs/issues/5437 |

### CODE REVIEW

File: cookbook/service_container/shared.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. index::
+    single: Service Container; Shared Services
+
+How to Define Not Shared Services
```
Comment (WouterJ) [inline cookbook/service_container/shared.rst]: "not shared", "non shared", "unshared", ...? (in other words: I'm not sure how to call this)

Comment (dosten) [inline cookbook/service_container/shared.rst]: IMO "non shared"

File: cookbook/service_container/shared.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    .. code-block:: php
+
+        $definition = new Definition('...');
+        $definition->setShared(false);
+
+        $container->setDefinition('app.some_not_shared_service', $definition);
+
+Now, whenever you call ``$container->get('app.some_not_shared_service')`` or
+inject this service, you'll recieve a new instance.
```
Comment (dosten) [inline cookbook/service_container/shared.rst]: Typo: recieve -> receive

File: cookbook/service_container/shared.rst (hunk 1 above)
Comment (xabbuh) [inline cookbook/service_container/shared.rst]: :+1: for "non shared"

File: cookbook/service_container/shared.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+How to Define Not Shared Services
+=================================
+
+.. versionadded:: 2.8
+    The ``shared`` setting was introduced in Symfony 2.8. Prior to Symfony 2.8,
+    you had to use the ``prototype`` scope.
+
+In the service container, all services are shared by default. This means that
+each time you retrieve the service, you'll get the *same* instance. This is
+often the behaviour you want, but in some cases, you might want to always get a
```
Comment (xabbuh) [inline cookbook/service_container/shared.rst]: behavior

File: cookbook/service_container/shared.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+*new* instance.
+
+In order to always get a new instance, set the ``shared`` setting to ``false``
+in your service definition:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        services:
```
Comment (xabbuh) [inline cookbook/service_container/shared.rst]: missing `# app/config/services.yml`

File: cookbook/service_container/shared.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            app.some_not_shared_service:
+                class: ...
+                shared: false
+                # ...
+
+    .. code-block:: xml
+
+        <services>
```
Comment (xabbuh) [inline cookbook/service_container/shared.rst]: missing `# app/config/services.xml`

File: cookbook/service_container/shared.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            <service id="app.some_not_shared_service" class="..." shared="false" />
+        </services>
+
+    .. code-block:: php
+
+        $definition = new Definition('...');
```
Comment (xabbuh) [inline cookbook/service_container/shared.rst]: missing `# app/config/services.`

Comment (xabbuh) [inline cookbook/service_container/shared.rst]: and missing `use` statement

**Comment by weaverryan** [comment]
Thanks Wouter! And thanks for the review - I've implemented the tweaks suggested.

**Pull request merged**